client/js/store/modules/auth.js

```diff
@@ -107,5 +106,25 @@ export default {
         );
       });
     },
+    requestStatus(context) {
+      return new Promise((resolve, reject) => {
+        io.socket.get('/user/status', {}, function handleResponse(resData, jwres) {
+          if (jwres.statusCode !== 200) {
+            context.commit('clearAuth');
+            return reject(new Error('Error getting user status'));
+          }
+          const { authenticated, id, username } = resData;
+          if (authenticated && username) {
+            context.commit('authSuccess', username);
+          }
+          return resolve(id);
+        });
+      });
+    },
```

Comment:
Vue usage of the new method to handle the user's status check, logging them in if they're valid and logged in on the server, or clearing the auth if they're not.